### ISSUE TITLE
Trouble with NSCollectionView usage

### ISSUE THREAD

**Issue by rudedogg** [tonyarnold/Differ#26]
I'm having some trouble using Differ with an NSCollectionView, specifically the function `animateItemsAndSectionChanges(oldData: , newData:)`

 Here's a quick example of the code I'm using:

```swift
struct MySection: Equatable {
  let name = "Test"
  var items = [String]()
}
```

```swift
func quickRefreshTest(oldData: [MySection], newData: [MySection]) {
  collectionView.animateItemAndSectionChanges(oldData: oldData, newData: newData)
}
```

But the compiler complains `Generic parameter 'T' could not be inferred`.

**Edit:** This doesn't seem to be a bug, I think it was just a compiler/generics/protocols difficulty. I'll post more details tomorrow

**Comment by tonyarnold** [comment]
If there's something I can add (including documentation) - please let me know. I'd love to know more about what you saw.

**Comment by rudedogg** [comment]
Thanks @tonyarnold 

I think the main issue I ran into was using Protocols as types in my datasource. That makes Equatable hard to implement, I had to use type erasure following https://khawerkhaliq.com/blog/swift-protocols-equatable-part-one/). I've since switched to passing a nested Array of `AnyEquatable{Protocol}` to the calls to `collectionView..animateItemAndSectionChanges(oldData: oldData, newData: newData)`, and the compiler is happy! It *seems* like using my struct above should work, but I think it's just a limitation of the Swift compiler.
